src/Bicep.Core/Syntax/SyntaxVisitor.cs

```diff
@@ -333,12 +332,12 @@ public virtual void VisitImportDeclarationSyntax(ImportDeclarationSyntax syntax)
 
         public virtual void VisitParameterAssignmentSyntax(ParameterAssignmentSyntax syntax)
         {
-            throw new NotImplementedException();
+            //TODO: unsure what implementation would mean here
```

Comment:
Yes, you want a default behavior here so that existing visitors can be used on parameters files. Visiting the child nodes is the right default behavior.

Reply:
Yes, exactly, visit child nodes/tokens in the correct order. The visitor needs to be able to recurse down to **all** syntax nodes.

If you don't do this, then anyone extending this visitor to visit specific node types will not be invoked for any of the nodes below your new syntax. For example, if someone wanted to visit all strings in a file, they would not see `mystring` under:
```bicep
param foo = {
  val = 'mystring'
}
```

Reply:
Similarly, make sure you implement the equivalent methods in the rewrite visitor - again following the pattern used elsewhere in the file:

https://github.com/Azure/bicep/blob/7286ae7631580fdb90c9524a2f6e4d8fbae948b6/src/Bicep.Core/Syntax/SyntaxRewriteVisitor.cs#L153-L161

Reply:
Seems like Cathy already implemented SyntaxVisitors but I will go ahead implement syntax rewrite visitors